pubsub.go

```diff
@@ -260,6 +264,8 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 		counter:               uint64(time.Now().UnixNano()),
 	}
 
+	ps.newPeersSema <- struct{}{}
```

Comment:
nit: I generally prefer the pattern where _writing_ takes the lock, and reading unlocks. Just means you can avoid this initialization part.